src/runtime/nitro/plugins/00-context.ts

```diff
@@ -0,0 +1,24 @@
+import { type HeaderMapper, getHeaderValueFromOptions, SECURITY_HEADER_NAMES } from "../../utils/headers"
+import { createRouter} from "radix3"
+import { defineNitroPlugin } from '#imports'
+
+export default defineNitroPlugin((nitroApp) => {
+    const router = createRouter()
+
+    nitroApp.hooks.hook('nuxt-security:headers', (route, headersConfig) => {
+        const headers: Record<string, string |false > = {}
+
+        for (const [header, headerOptions] of Object.entries(headersConfig)) {
+            headers[SECURITY_HEADER_NAMES[header]] = headerOptions === false ? false : getHeaderValueFromOptions(header as HeaderMapper, headerOptions as any)
+        }
+
+        router.insert(route, headers)
+    })
+
+    nitroApp.hooks.hook('request', (event) => {
+        event.context.security = event.context.security || {}
```

Comment:
I'd use the module name to avoid conflicts

Reply:
i used security because it's the module `configKey`, but yeah we can change it to `nuxtSecurity` !

Reply:
I think if ends users will use this context object it could make sense with this name, otherwise if it's just internal module name is better.

Reply:
I would stick to `security` instead of `nuxt-security` purely because this is a config key for both nuxt.config.ts file and for route rules so it would be better IMHO to keep this consistent :)